internal/utils/chaincode.go

```diff
@@ -0,0 +1,43 @@
+/*
```

Comment:
More high-level questions: It's not clear to me what distinguishes `internal` from `common`.  What is the (clear) semantic difference between the two?  Otherwise i think we might eventually pick one or the other (as otherwise you have to now always look at two place to figure out whether some exists or not ...)

Reply:
`common` currently only contains c/cpp code, where `internal` only has go code. Also, `internal` is a special folder that prevents external projects from importing any packages from them.

I think we should maintain this separation of c and go code. Maybe find better names?

Reply:
Not immediately clear to me why keeping directory based on language separate, in particular on the top-level and with a name which completely obfuscates that. What do you see as an advantage to keep it separate?  The only thing which could come in mind is issues i encountered with vendoring and cgo (e.g., `go mod vendor` does only copy subsets of code into the vendor directories where right now only c-code residing in same directory of (referenced) go-code and eventually even that might go from various discussion ...) but for that one probably would have to restricture it globally with all go and non-go code in seperate sub-dirs if not even repos. (e.g., c-code would be in separate library and would have to be built _and installed_ before building go and cgo references.

Reply:
yes, my concern is mostly related to cgo. This is also the reason why I initially introduced `ecc` and `ecc_enclave`.